src/test/transaction_tests.cpp

```diff
@@ -149,17 +154,20 @@ unsigned int FillFlags(unsigned int flags)
 
     // WITNESS implies P2SH (and transitively CLEANSTACK implies P2SH)
     if (flags & SCRIPT_VERIFY_WITNESS) flags |= SCRIPT_VERIFY_P2SH;
+    Assert(IsValidFlagCombination(flags));
     return flags;
 }
 
-// Return valid flags that are all except one flag for each flag
-std::vector<unsigned int> ExcludeIndividualFlags(unsigned int flags)
+// Exclude each possible script verify flag from flags. Returns a list of these flag combinations
+// that are valid and without duplicates. For example: if flags=1111 and there are 4 possible flags,
+// this should return the vector [0111, 1011, 1101, 1110].
+std::set<unsigned int> ExcludeIndividualFlags(unsigned int flags)
 {
-    std::vector<unsigned int> flags_combos;
+    std::set<unsigned int> flags_combos;
     for (unsigned int i = 0; i < mapFlagNames.size(); ++i) {
         const unsigned int flags_excluding_i = TrimFlags(flags & ~(1U << i));
```

Comment:
I still don't think this is right. A std::map doesn't order its elements based on the order they were inserted. 

Even if it did, the values aren't guaranteed to never miss a power of two.

Reply:
Sorry, what do you mean? The map `mapFlagNames` is just to identify the flags from strings in the json files. I guess if there wasn't a script verify flag for `1 << i` for some `i`, it would get ignored by the interpreter if we added it. So as long as we have the leftmost flag we should be okay?

Reply:
Let's assume mapFlagNames is only a single entry `{"EXAMPLE", 1U << 4}` (because for some reason other flags are no longer needed and have been removed).

The loop will only cover `1U << 1`, a non-existent flag.

Reply:
Ah true. Is there a way of checking whether an unsigned int corresponds to a script verify flag?

Reply:
Just iterate over the map and use the value of the key-value pair?